vta/tests/python/tsim/tsim_load_store.py

```diff
@@ -0,0 +1,99 @@
+"""
+Load and store test
+======================
+"""
+
+######################################################################
+# RPC Setup
+# ---------
+# We start by programming the Pynq's FPGA and building its RPC runtime
+# as we did in the VTA introductory tutorial.
+
+from __future__ import absolute_import, print_function
+
+import os
+import tvm
+import vta
+import numpy as np
+from tvm import rpc
+from tvm.contrib import util
+from vta.testing import simulator
+
+# Load VTA parameters from the vta/config/vta_config.json file
+env = vta.get_env()
+
+# We read the Pynq RPC host IP address and port number from the OS environment
+host = os.environ.get("VTA_PYNQ_RPC_HOST", "192.168.2.99")
+port = int(os.environ.get("VTA_PYNQ_RPC_PORT", "9091"))
+
+# We configure both the bitstream and the runtime system on the Pynq
+# to match the VTA configuration specified by the vta_config.json file.
+if env.TARGET == "pynq":
+
+    # Make sure that TVM was compiled with RPC=1
+    assert tvm.module.enabled("rpc")
+    remote = rpc.connect(host, port)
+
+    # Reconfigure the JIT runtime
+    vta.reconfig_runtime(remote)
+
+    # Program the FPGA with a pre-compiled VTA bitstream.
+    # You can program the FPGA with your own custom bitstream
+    # by passing the path to the bitstream file instead of None.
+    vta.program_fpga(remote, bitstream=None)
+
+# In simulation mode, host the RPC server locally.
+elif env.TARGET == "sim" or env.TARGET == "tsim":
+    remote = rpc.LocalSession()
+
```

Comment:
do we need to handle/report such case that env.TARGET not in ["pynq", "sim", "tsim"]? seems like currently once such things happen remote would be a invalid value.

Reply:
Could you clarify a bit the question please?

Two things that might be related:

1. There are going to be two simulation targets now, one for the functional simulator "sim" and another for the cycle-accurate-simulator or "tsim"

2. The tests in this directory will be eventually merged to a single place and will be configured depending on the target. The reason why they are here is to show/discuss how we are going to handle them.

Reply:
this question is talking about a corner case, when env.TARGET is a wrong value for example "zcu", there is no code branch to handle for 'remote' variable set,  then 'remote' would be a uninitialized/invalid value, and cause crash at line 82 when doing 'remote.load' ,   is this a intend behavior? or we should 'handle such issue'(report error) in this unit test code?

Reply:
Oh I see, thanks for clarifying. I thinks when you specify a bad target in the config.json file, i.e. "zcu", the script will definitely going to crash as you said. I think it is the intended behavior and not particular to these tests. I think the VTA tutorials use that see [here](https://github.com/dmlc/tvm/blob/master/vta/tutorials/matrix_multiply.py).

Do we want to address this differently? @tmoreau89